File: VB/DXSample/DataHelper.vb
Imports System
Imports System.Collections.ObjectModel
Imports System.Windows.Media
Imports System.Windows.Media.Imaging
Imports System.Drawing
Imports System.ComponentModel

Namespace DXSample

    Public Module DataHelper

        Public Function GetData1() As ObservableCollection(Of ExampleObject)
            Dim collection As ObservableCollection(Of ExampleObject) = New ObservableCollection(Of ExampleObject)() From {New ExampleObject With {.Name = "Basket", .ImageUri = "Images/Basket.png"}, New ExampleObject With {.Name = "Check", .ImageUri = "Images/Check.png"}, New ExampleObject With {.Name = "Customer", .ImageUri = "Images/Customer.png"}}
            Return collection
        End Function

        Public Function GetData2() As ObservableCollection(Of ExampleObject)
            Dim collection As ObservableCollection(Of ExampleObject) = New ObservableCollection(Of ExampleObject)() From {New ExampleObject With {.Name = "Folder", .ImageUri = "Images/Folder.png"}, New ExampleObject With {.Name = "Key", .ImageUri = "Images/Key.png"}, New ExampleObject With {.Name = "Home", .ImageUri = "Images/Home.png"}}
            Return collection
        End Function
    End Module

    Public Class ExampleObject
        Inherits DevExpress.Mvvm.BindableBase

        ' Fields...
        Private _ImageUri As String

        Private _Name As String

        Public Property Name As String
            Get
                Return _Name
            End Get

            Set(ByVal value As String)
                SetProperty(_Name, value, Function() Name)
            End Set
        End Property

        Public Property ImageUri As String
            Get
                Return _ImageUri
            End Get

            Set(ByVal value As String)
                SetProperty(_ImageUri, value, Function() ImageUri)
                RaisePropertiesChanged("ImageSource")
            End Set
        End Property

        Public ReadOnly Property ImageSource As ImageSource
            Get
                Return If(String.IsNullOrEmpty(ImageUri), Nothing, New BitmapImage(New Uri("/DXSample;component/" & ImageUri, UriKind.Relative)))
            End Get
        End Property
    End Class
End Namespace

File: CS/DXSample/DataHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections.ObjectModel;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Drawing;
using System.ComponentModel;

namespace DXSample
{
    public static class DataHelper
    {
        public static ObservableCollection<ExampleObject> GetData1()
        {
            ObservableCollection<ExampleObject> collection = new ObservableCollection<ExampleObject>()
            {
                new ExampleObject{Name = "Basket", ImageUri ="Images/Basket.png"},
                new ExampleObject{Name = "Check", ImageUri ="Images/Check.png"},
                new ExampleObject{Name = "Customer", ImageUri ="Images/Customer.png"},
            };
            return collection;
        }
        public static ObservableCollection<ExampleObject> GetData2()
        {
            ObservableCollection<ExampleObject> collection = new ObservableCollection<ExampleObject>()
            {
                new ExampleObject{Name = "Folder", ImageUri ="Images/Folder.png"},
                new ExampleObject{Name = "Key", ImageUri ="Images/Key.png"},
                new ExampleObject{Name = "Home", ImageUri ="Images/Home.png"},
            };
            return collection;
        }
    }
    public class ExampleObject : INotifyPropertyChanged {
        // Fields...
        private string _ImageUri;
        private string _Name;

        public string Name {
            get { return _Name; }
            set {
                if (_Name == value)
                    return;
                _Name = value;
                NotifyPropertyChanged("Name");
            }
        }

        public string ImageUri {
            get { return _ImageUri; }
            set {
                if (_ImageUri == value)
                    return;
                _ImageUri = value;
                NotifyPropertyChanged("ImageUri");
                NotifyPropertyChanged("ImageSource");
            }
        }

        public ImageSource ImageSource {
            get {
                return string.IsNullOrEmpty(ImageUri) ? null : new BitmapImage(new Uri("/DXSample;component/" + ImageUri, UriKind.Relative));
            }
        }
        
        public event PropertyChangedEventHandler PropertyChanged;
        protected void NotifyPropertyChanged(String info) {
            if (PropertyChanged != null) {
                PropertyChanged(this, new PropertyChangedEventArgs(info));
            }
        }
    }
}
